examples/Misc/UseCustomClientTimeouts.php

```diff
@@ -0,0 +1,190 @@
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+            exit(1);
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        self::serverStreamingCall($googleAdsClient, $customerId);
+        self::unaryCall($googleAdsClient, $customerId);
+    }
+
+    /**
+     * Makes a server streaming call using a custom client timeout.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    private static function serverStreamingCall(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates a query that retrieves all campaign IDs.
+        $query = 'SELECT campaign.id FROM campaign';
+
+        $output = '';
+        try {
+            // Issues a search stream request.
+            /** @var GoogleAdsServerStreamDecorator $stream */
+            $stream = $googleAdsServiceClient->searchStream(
+                $customerId,
+                $query,
+                ['timeoutMillis' => self::CUSTOM_CLIENT_TIMEOUT]
+            );
+            // Iterates over all rows in all messages and collects the campaign IDs.
+            foreach ($stream->iterateAllElements() as $googleAdsRow) {
+                /** @var GoogleAdsRow $googleAdsRow */
+                $output .= ' ' . $googleAdsRow->getCampaign()->getId();
+            }
+            print 'The server streaming call completed before the timeout.' . PHP_EOL;
+        } catch (ApiException $exception) {
+            if ($exception->getStatus() === ApiStatus::DEADLINE_EXCEEDED) {
+                print 'The server streaming call did not complete before the timeout.' . PHP_EOL;
+            } else {
+                // Bubbles up if the exception is not about timeout.
+                throw $exception;
+            }
+        } finally {
+            print 'All campaign IDs retrieved:' . ($output ?: ' None') . PHP_EOL;
+        }
+    }
+
+    /**
+     * Makes an unary call using a custom client timeout.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    private static function unaryCall(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates a query that retrieves all campaign IDs.
+        $query = 'SELECT campaign.id FROM campaign';
+
+        $output = '';
+        try {
+            // Issues a search request.
```

Comment:
Ditto. There are some settings that are different from normal code examples.
We probably can explain a bit more here.